lib/screen/onboarding/store/on_boarding_store.dart

```diff
@@ -57,6 +59,35 @@ abstract class _OnBoardingStore with Store {
 
   @observable
   bool isLoading = false;
+  static const List<String> scopes = <String>[
+    'email',
+    'https://www.googleapis.com/auth/contacts.readonly',
+  ];
+
+  GoogleSignIn _googleSignIn = GoogleSignIn(
+    serverClientId: BuildConfig.oAuthClientId,
+    scopes: scopes,
+  );
+
+  @action
+  Future<bool> loginWithGoogle() async {
+    try {
+      await _googleSignIn.signIn();
+
+      final googleAuth = await _googleSignIn.currentUser!.authentication;
+      final accessToken = googleAuth.accessToken;
+      final idToken = googleAuth.idToken;
+      Supabase.instance.client.auth.signInWithIdToken(
+        provider: OAuthProvider.google,
+        idToken: idToken!,
+        accessToken: accessToken,
+      );
+      return true;
+    } catch (error, stackTrace) {
+      log(error.toString(), stackTrace: stackTrace);
+      return false;
+    }
+  }
```

Comment:
```suggestion
  @action
  Future<bool> loginWithGoogle() async {
    try {
      await _googleSignIn.signIn();
      if (_googleSignIn.currentUser == null) return false;
      final googleAuth = await _googleSignIn.currentUser!.authentication;
      final accessToken = googleAuth.accessToken;
      final idToken = googleAuth.idToken;
      if (idToken == null) return false;
      Supabase.instance.client.auth.signInWithIdToken(
        provider: OAuthProvider.google,
        idToken: idToken,
        accessToken: accessToken,
      );
      return true;
    } catch (error, stackTrace) {
      log(error.toString(), stackTrace: stackTrace);
      return false;
    }
  }
```

Add a null check for the `_googleSignIn.currentUser` and `idToken` to avoid NPE.